This PR proposes changes that would break the pipeline definition:
```
landscaper-upload-test-results-2: Traceback (most recent call last):
  File "/usr/lib/python3.10/site-packages/concourse/replicator.py", line 140, in render
    definition_descriptor = self._render(definition_descriptor)
  File "/usr/lib/python3.10/site-packages/concourse/replicator.py", line 223, in _render
    raise RuntimeError(
RuntimeError: An exception occured when rendering pipeline 'landscaper-upload-test-results-2': 

Traceback (most recent call last):
  File "/usr/lib/python3.10/site-packages/concourse/replicator.py", line 215, in _render
    definition_descriptor.pipeline = t.render(
  File "/usr/lib/python3.10/site-packages/mako/template.py", line 439, in render
    return runtime._render(self, self.callable_, args, data)
  File "/usr/lib/python3.10/site-packages/mako/runtime.py", line 874, in _render
    _render_context(
  File "/usr/lib/python3.10/site-packages/mako/runtime.py", line 916, in _render_context
    _exec_template(inherit, lclcontext, args=args, kwargs=kwargs)
  File "/usr/lib/python3.10/site-packages/mako/runtime.py", line 943, in _exec_template
    callable_(context, *args, **kwargs)
  File "memory:0x7f7a0386c400", line 652, in render_body
    ${job(variant)}
  File "memory:0x7f7a0386c400", line 0, in job
    
  File "memory:0x7f7a0386c400", line 640, in render_job
    ${step(4, job_variant, job_variant.step(step_name))}
  File "/usr/lib/python3.10/site-packages/concourse/model/job.py", line 238, in step
    raise ValueError(f'no such step: {name=}')
ValueError: no such step: name='run_integration_test'


```

--- a/.ci/integration-test
+++ b/.ci/integration-test
@@ -13,10 +13,16 @@ set -o errexit
 set -o nounset
 set -o pipefail
 
+CURRENT_PATH="$(pwd)"
+cd "${INTEGRATION_TEST_PATH}"
+FULL_INTEGRATION_TEST_PATH="$(pwd)"
+export FULL_INTEGRATION_TEST_PATH
+cd "${CURRENT_PATH}"
+
 SOURCE_PATH="$(dirname $0)/.."
 cd "${SOURCE_PATH}"
 SOURCE_PATH="$(pwd)"
 export SOURCE_PATH
 
-./.ci/int-test "1"
+./.ci/int-test "1" | tee $FULL_INTEGRATION_TEST_PATH/ttt.log
 

--- a/.ci/pipeline_definitions
+++ b/.ci/pipeline_definitions
@@ -108,6 +108,7 @@ landscaper:
           trait_depends:
           - publish
           image: 'golang:1.18.5-alpine3.16'
+          output_dir: 'integration_test'
         update_release:
           inputs:
             INTEGRATION_TEST_PATH: integration_test_path

--- a/.ci/pipeline_definitions
+++ b/.ci/pipeline_definitions
@@ -68,7 +68,7 @@ landscaper:
               - committers
               - codeowners
       steps:
-        integration-test:
+        run_integration_test:
           execute:
           - integration-test
           trait_depends:
@@ -102,7 +102,7 @@ landscaper:
           - ref_template: 'refs/tags/apis/{VERSION}'
           - ref_template: 'refs/tags/controller-utils/{VERSION}'
       steps:
-        integration-test:
+        run_integration_test:
           execute:
           - integration-test
           trait_depends:

--- a/.ci/pipeline_definitions
+++ b/.ci/pipeline_definitions
@@ -74,11 +74,12 @@ landscaper:
           trait_depends:
           - publish
           image: 'golang:1.18.5-alpine3.16'
+          output_dir: 'integration_test'
     pull-request:
       steps:
         run_integration_test:
-          #depends:
-          #- publish
+          depends:
+          - publish
           image: 'golang:1.18.5-alpine3.16'
           execute:
           - integration-test-new
@@ -107,3 +108,12 @@ landscaper:
           trait_depends:
           - publish
           image: 'golang:1.18.5-alpine3.16'
+        update_release:
+          inputs:
+            INTEGRATION_TEST_PATH: integration_test_path
+          execute:
+          - update_release.py
+          trait_depends:
+          - release
+          depends:
+          - run_integration_test

--- a/.ci/update_release.py
+++ b/.ci/update_release.py
@@ -1,0 +1,46 @@
+#!/usr/bin/env python3
+
+import pathlib
+import util
+import os
+
+from github.util import GitHubRepositoryHelper
+
+VERSION_FILE_NAME = 'VERSION'
+
+repo_owner_and_name = util.check_env('SOURCE_GITHUB_REPO_OWNER_AND_NAME')
+repo_dir = util.check_env('MAIN_REPO_DIR')
+
+repo_owner, repo_name = repo_owner_and_name.split('/')
+
+repo_path = pathlib.Path(repo_dir).resolve()
+
+version_file_path = repo_path / VERSION_FILE_NAME
+
+version_file_contents = version_file_path.read_text()
+
+cfg_factory = util.ctx().cfg_factory()
+github_cfg = cfg_factory.github('github_com')
+
+github_repo_helper = GitHubRepositoryHelper(
+    owner=repo_owner,
+    name=repo_name,
+    github_cfg=github_cfg,
+)
+
+gh_release = github_repo_helper.repository.release_from_tag(version_file_contents)
+
+
+try:
+    os.environ['INTEGRATION_TEST_PATH']
+except KeyError:
+    print("No integration test output path found. Output will not be added to release")
+else:
+    integration_test_path = util.check_env('INTEGRATION_TEST_PATH')
+    integration_test_path = pathlib.Path(integration_test_path).resolve()
+    integration_test_path = integration_test_path / "ttt.log"
+    gh_release.upload_asset(
+        content_type='text/plain',
+        name=f'integration-test-result-{version_file_contents}.txt',
+        asset=integration_test_path.open(mode='rb'),
+    )

